### PR TITLE
Add ImportAlt to Arbitrary instance, fix isNormalized and diff

### PR DESCRIPTION
**Pull request by sjakobi**
This is built on top of https://github.com/dhall-lang/dhall-haskell/pull/1266.

Context: https://github.com/dhall-lang/dhall-haskell/issues/1256#issuecomment-526829515

### CODE REVIEW

File: dhall/src/Dhall/Diff.hs [hunk 1]
```diff
@@ -745,7 +754,22 @@ diffAnnotatedExpression l r =
     diffOperatorExpression l r
 
 diffOperatorExpression :: (Eq a, Pretty a) => Expr Void a -> Expr Void a -> Diff
-diffOperatorExpression = diffOrExpression
+diffOperatorExpression = diffImportAltExpression
+
+diffImportAltExpression :: (Pretty a, Eq a) => Expr Void a -> Expr Void a -> Diff
+diffImportAltExpression l@(ImportAlt {}) r@(ImportAlt {}) =
+    enclosed' "    " (operator "?" <> "  ") (docs l r)
```
Comment (sjakobi) [inline dhall/src/Dhall/Diff.hs]: Am I using the right amount of space here? It seems to vary between the various `diff*Expression`s but I'm not sure why.

File: dhall/src/Dhall/Diff.hs [hunk 2]
```diff
@@ -745,7 +754,22 @@ diffAnnotatedExpression l r =
...
+  where
+    docs (ImportAlt aL bL) (ImportAlt aR bR) =
+        Data.List.NonEmpty.cons (diffOrExpression aL aR) (docs bL bR)
```
Comment (sjakobi) [inline dhall/src/Dhall/Diff.hs]: Is the use of `diffOrExpression` correct here? In general these functions seem to use the diff for next-higher precedence here, but a lot of them also deviate.

File: dhall/src/Dhall/Diff.hs (hunk 1 above)
Comment (Gabriel439) [reply dhall/src/Dhall/Diff.hs]: All indentation (whether pretty-printing or diffing) is a multiple of two spaces, so if the operator is one character long (like `?`) then the diff pads the left margin to two space:

```
␣␣e₀
?␣e₁
```

... but if the operator is two characters long (like `||`) then the diff pads the left margin to four spaces:

```
␣␣␣␣e₀
||␣␣e₁
```

So in this case what you want is:

```haskell
enclosed' "  " (operator "?" <> " ") (docs l r)
```

File: dhall/src/Dhall/Diff.hs (hunk 2 above)
Comment (Gabriel439) [reply dhall/src/Dhall/Diff.hs]: Your use of `diffOrExpression` here is correct and the deviations from this pattern are bugs

File: dhall/src/Dhall/Diff.hs (hunk 1 above)
Comment (sjakobi) [reply dhall/src/Dhall/Diff.hs]: Thanks! I've added your explanation as a comment on `diffOperatorExpression.

**Pull request merged**